src/main/java/Duke.java

```diff
@@ -1,10 +1,262 @@
+                markTask(Integer.parseInt(arg2),
+                        false);
+                break;
+            case "todo":
+                try {
+                    addTodo(arg2);
+                } catch (DukeException e) {
+                    handleAddTodoException();
+                }
+                break;
+            case "deadline":
+                try {
+                    addDeadline(arg2);
+                } catch (DukeException e) {
+                    handleAddDeadException(e);
+                }
+                break;
+            case "event":
+                try {
+                    addEvent(arg2);
+                } catch (DukeException e) {
+                    handleAddEventException(e);
+                }
+                break;
+            case "delete":
+                deleteTask(Integer.valueOf(arg2));
+                break;
+            default:
+                try {
+                    addTask(input);
+                } catch (IllegalArgsTypeException e) {
+                    handleAddTaskException();
+                }
+            }
+            input = sc.nextLine();
+        }
+    }
+
+    private static void addTask(String taskDescription) throws IllegalArgsTypeException {
+        if (Arrays.asList(ILLEGAL_ARGS).contains(taskDescription)) {
+            throw new IllegalArgsTypeException();
+        }
+        Task new_task = new Task(taskDescription);
+        tasks.add(new_task);
+        System.out.println(">>>Added: " + new_task);
+        dumpTask(new_task);
+        showTaskCount();
+    }
+
+    private static void listTask() {
+        if (tasks.size() == 0) {
+            System.out.println(">>>No Current Tasks.");
+        } else {
+            System.out.println(">>>Current Tasks:");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.print(">>>" + (i + 1) + ".");
+                System.out.println(tasks.get(i));
+            }
+        }
+        showSeparator();
+    }
+
+    private static void markTask(int taskId, boolean setMark) {
+        if (taskId <= 0 || taskId > tasks.size()) {
+            System.out.println(">>>Pls Enter the Right TaskId!");
+        }
+        else {
+            tasks.get(taskId-1).setMarked(setMark);
+            if (setMark) {
+                System.out.println(">>>Nice! I've marked this task as done:");
+            }
+            else {
+                System.out.println(">>>OK, I've marked this task as not done yet:");
+            }
+            System.out.println(tasks.get(taskId-1));
+        }
+        showSeparator();
+    }
+
+    private static void deleteTask(int num) {
+        if (num <= 0 || num > tasks.size()) return;
+        Task tmp_task = tasks.get(num - 1);
+        tasks.remove(num - 1);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(">>>" + tmp_task);
+        showTaskCount();
+    }
+
+    private static void dumpTask(Task task) {
```

Comment:
Naming of this method could also be improved, as just `dumpTasks` might confuse the reader with `deleteTask`. A better name could be `saveTasksIntoFile` or `dumpTasksIntoFile`.